crates/evm/src/trace/identifier/etherscan.rs

```diff
@@ -60,10 +60,14 @@ impl EtherscanIdentifier {
     /// Etherscan and compiles them locally, for usage in the debugger.
     pub async fn get_compiled_contracts(
         &self,
-    ) -> eyre::Result<(BTreeMap<ArtifactId, String>, BTreeMap<ArtifactId, ContractBytecodeSome>)>
-    {
+    ) -> eyre::Result<(
+        // TODO should use `ContractSources` but has circular import.
+        // Maybe move it lower
+        HashMap<String, HashMap<u32, (String, ContractBytecodeSome)>>,
+        BTreeMap<ArtifactId, ContractBytecodeSome>,
+    )> {
         let mut compiled_contracts = BTreeMap::new();
-        let mut sources = BTreeMap::new();
+        let mut sources = HashMap::new();
```

Comment:
This shuffles ordering—is ordering an issue here?

Reply:
We don't need ordering, so there should be better performances with a `HashMap`
Can leave as-is if not appriopriate for this PR